extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/videos/GoogleVideosImporter.java

```diff
@@ -127,27 +122,26 @@ void importSingleVideo(
     }
 
     String uploadToken =
-            getOrCreateVideosInterface(authData).uploadVideoContent(inputStream, filename);
+        getOrCreateVideosInterface(authData).uploadVideoContent(inputStream, filename);
 
     NewMediaItem newMediaItem = new NewMediaItem(filename, uploadToken);
 
     NewMediaItemUpload uploadItem =
-            new NewMediaItemUpload(null, Collections.singletonList(newMediaItem));
+        new NewMediaItemUpload(null, Collections.singletonList(newMediaItem));
 
-    executor.executeAndSwallowIOExceptions(
-            inputVideo.getDataId(),
-            inputVideo.getName(),
-            () -> getOrCreateVideosInterface(authData).createVideo(uploadItem));
+    final NewMediaItemResult result =
+        getOrCreateVideosInterface(authData).createVideo(uploadItem).getResults()[0];
+    return result.getMediaItem().getId();
   }
 
   private synchronized GoogleVideosInterface getOrCreateVideosInterface(
-          TokensAndUrlAuthData authData) {
+      TokensAndUrlAuthData authData) {
     return videosInterface == null ? makeVideosInterface(authData) : videosInterface;
   }
 
   private synchronized GoogleVideosInterface makeVideosInterface(TokensAndUrlAuthData authData) {
     Credential credential = credentialFactory.createCredential(authData);
-    GoogleVideosInterface videosInterface = new GoogleVideosInterface(credential, this.jsonFactory);
+    videosInterface = new GoogleVideosInterface(credential, this.jsonFactory);
```

Comment:
Should we just inline with the return statement below?

Reply:
This is setting the field on the class rather than a local variable so that it will be reused.